hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/NameNodeProxiesClient.java

```diff
@@ -349,6 +349,13 @@ public static ClientProtocol createProxyWithAlignmentContext(
       boolean withRetries, AtomicBoolean fallbackToSimpleAuth,
       AlignmentContext alignmentContext)
       throws IOException {
+    if (conf.getBoolean(HdfsClientConfigKeys.DFS_RBF_OBSERVER_READ_ENABLE,
```

Comment:
What was the original behaviour where someone passed in `null` to this function?

Reply:
If someone passes in null, then the client will not echo back alignment state to the router, which it turn makes the router always forward these calls to the active namenode.